### ISSUE TITLE
Misplaced error span in TS 5.1 beta

### ISSUE THREAD

**Issue by danvk** [microsoft/TypeScript#53932]
# Bug Report

### 🔎 Search Terms

- 5.1 beta

### 🕗 Version & Regression Information

- This changed between versions 5.0.4 and 5.1.0-beta

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=5.1.0-dev.20230420#code/GYVwdgxgLglg9mABMOcAUBKAXIgCgJzgFsYBnAUwB4wQiAjc-APkQG8AoRRfcqEfJAEYADAG52AX3ZA)

### 💻 Code

```ts
function foo(): Promise<number> {
  return 10;
}
```

### 🙁 Actual behavior

With TypeScript 5.1 beta, the error is reported on the `return` keyword itself:

<img width="503" alt="image" src="https://user-images.githubusercontent.com/98301/233425402-f4e4b1fa-1879-4a2c-aac5-5e31ebd6544f.png">

### 🙂 Expected behavior

The error span makes more sense in TypeScript 5.0.4 and earlier, where it includes the value:

<img width="501" alt="image" src="https://user-images.githubusercontent.com/98301/233425655-0d7ad449-d0e0-4afd-88b4-06beeb774971.png">

**Comment by fatcerberus** [comment]
This was an intentional change.  See https://github.com/microsoft/TypeScript/issues/52934 for rationale (in particular the screenshot at the top of that issue)

**Comment by danvk** [comment]
Yeah OK, hard to argue with that screenshot :)
